### DISCUSSION TITLE
Why is this extension using the Visual Studio version of Doxygen doc hints?

### DISCUSSION THREAD

**Discussion by BlueStaggo** [microsoft/vscode-cpptools#7912] [General]
We could be more ahead if this extension used a different method. This extension doesn't even support things like the `@link` tag!
![image](https://user-images.githubusercontent.com/71090710/128066227-9abc2c8b-a0ea-41a7-95ec-905de6bc78bb.png)
![image](https://user-images.githubusercontent.com/71090710/128066305-7fc6164f-fcd2-40d6-a4d9-3f2579f0a36d.png)
![image](https://user-images.githubusercontent.com/71090710/128066429-929b8dc7-10dc-4c97-8615-4c19c4bdf72f.png)
![image](https://user-images.githubusercontent.com/71090710/128066514-b5055cfb-2ba8-4fe0-a87b-7652e934586c.png)
These are just a few examples of many from GLFW that are broken. I heared that this extension replicates Visual Studio's method of generating these doc hints:
> _sean-mcmanus on #6650:_
> Related to #6399 . Behavior matches what __VS__ shows.
> 
> michelleangela Do you know how to deal with this?

> _sean-mcmanus on #6084:_
> devabhishekpal That is implemented in our shared code with __VS__, not in our our open source TypeScript.

> _michelleangela on #6084:_
> This is not fixed yet. The code for this is common between the C++ extension and C++ language service on __Visual Studio__. So once the issue ([tracked by feedback ticket](https://developercommunity.visualstudio.com/content/problem/1081005/%E6%B3%A8%E9%87%8A%E4%B8%AD%E5%A6%82%E6%9E%9C%E5%87%BA%E7%8E%B0%E4%BA%86%E5%AD%97%E7%AC%A6-0-%E4%BC%9A%E5%AF%BC%E8%87%B4%E5%AD%97%E7%AC%A6-0-%E4%B9%8B%E5%90%8E%E7%9A%84%E6%B3%A8%E9%87%8A%E6%97%A0%E6%B3%95%E6%98%BE%E7%A4%BA.html)) is addressed for Visual Studio, the C++ extension will be able to port over the fix.

So what's this talk about using Visual Studio's language service? If Visual Studio's Doxygen doc hints are so bad, why don't we make our own? Is it too complicated or difficult or is it impossible to do such a thing? Also, apologies if I am getting the name wrong.

**Comment by sean-mcmanus** [comment]
Yeah, our implementation backed for VS Code doc comments is shared with VS. We have some existing issues tracking supporting more doxygen tags and stuff.

**Reply by BlueStaggo** [reply]
A problem is that I can't find an extension that implements their own method for doc hints instead of using Visual Studio. Is that possible? The closest I can find is "Doxygen Previewer", but that doesn't work sometimes and it doesn't affect doc hints.

**Reply by sean-mcmanus** [reply]
I'm not aware of an alternative doc hint provider for VS Code.
